src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.cpp

```diff
@@ -218,10 +218,65 @@ size_t ShenandoahOldGeneration::unexpend_promoted(size_t decrement) {
   return Atomic::sub(&_promoted_expended, decrement);
 }
 
-size_t ShenandoahOldGeneration::get_promoted_expended() {
+size_t ShenandoahOldGeneration::get_promoted_expended() const {
   return Atomic::load(&_promoted_expended);
 }
 
+bool ShenandoahOldGeneration::can_allocate(const ShenandoahAllocRequest &req) const {
+  assert(req.type() != ShenandoahAllocRequest::_alloc_gclab, "GCLAB pertains only to young-gen memory");
+
+  const size_t requested_bytes = req.size() * HeapWordSize;
+  if (can_promote(requested_bytes)) {
+    // The promotion reserve should be able to accommodate this request. The request
+    // might still fail if alignment with the card table increases the size. The request
+    // may also fail if the heap is badly fragmented and the free set cannot find room for it.
+    return true;
+  }
+
+  if (req.type() == ShenandoahAllocRequest::_alloc_plab) {
+    // The promotion reserve is exhausted. Check if we still have room for evacuations.
```

Comment:
We don't KNOW promotion reserve is entirely exhausted, just that what remains in promotion reserve is less than requested bytes.

Reply:
Fixed comment.